timestamp-impl/src/main/java/com/palantir/timestamp/PersistentTimestampService.java

```diff
@@ -137,10 +129,28 @@ public void run() {
         }
     }
 
-    private static boolean isAllocationRequired(long lastVal, long upperLimit) {
+    private boolean isAllocationRequired(long lastVal, long upperLimit) {
+        // we allocate new timestamps if we have less than half of our allocation buffer left
+        // or we haven't allocated timestamps in the last sixty seconds. The latter case
+        // exists in order to log errors faster against the class of bugs where your
+        // timestamp limit changed unexpectedly (usually, multiple TS against the same DB)
+        return exceededUpperLimit(lastVal, upperLimit)
+                || exceededHalfOfBuffer(lastVal, upperLimit)
+                || exceededLastAllocationTime();
+    }
+
+    private boolean exceededHalfOfBuffer(long lastVal, long upperLimit) {
         return (upperLimit - lastVal) <= ALLOCATION_BUFFER_SIZE / 2;
     }
 
+    private boolean exceededUpperLimit(long lastVal, long upperLimit) {
```

Comment:
I think this is redundant.  Whenever this is true exceededHalfOfBuffer will also be true.